jscomp/ext/ext_pp.ml

```diff
@@ -29,66 +29,98 @@ end
 
 let indent_length = String.length L.indent_str
 
+type kind =
+  | Channel of out_channel
+  | Buffer of Buffer.t
 type t = {
-  output_string : string -> unit;
-  output_char : char -> unit;
-  flush : unit -> unit;
+  kind : kind;
   mutable indent_level : int;
+  mutable column : int;
+  mutable line : int;
   mutable last_new_line : bool;
-      (* only when we print newline, we print the indent *)
+  (* only when we print newline, we print the indent *)
+}
+
+let output_string t s =
+  (match t.kind with
+  | Channel chan -> output_string chan s
+  | Buffer buf -> Buffer.add_string buf s);
+  let (new_line, new_column) =
+    Ext_string.fold_left 
+      (fun (line, column) char ->
+          if char = '\n' then
```

Comment:
Are there other line endings we should support? e.g. `\r`

Reply:
\r is not a valid line ending outside of the original Mac OS, it is always \r\n or just \n, which this handles both